social_engagement/handlers.py

```diff
@@ -11,107 +9,146 @@
     thread_deleted,
     comment_deleted,
     thread_voted,
-    comment_voted,
     thread_followed,
     thread_unfollowed,
+    thread_or_comment_flagged,
 )
 import lms.lib.comment_client as cc
-from social_engagement.tasks import task_update_user_engagement_score
+from social_engagement.tasks import task_handle_change_after_signal
 
 
 @receiver(thread_deleted)
-@receiver(comment_deleted)
 @receiver(thread_voted)
-@receiver(comment_voted)
-def thread_comment_signal_handler(sender, **kwargs):  # pylint: disable=unused-argument
+def thread_signal_handler(sender, **kwargs):  # pylint: disable=unused-argument
     """
-    Updates user social engagement score
+    Updates user social engagement score for deleted and voted thread (or voted comment).
+    TODO: And also, for some reason, on voted comment.
     """
-    post = kwargs['post']
-    course_id = getattr(post, 'course_id', None)
-    user_id = getattr(post, 'user_id', None)
-    social_engagement_enabled = settings.FEATURES.get('ENABLE_SOCIAL_ENGAGEMENT', False)
-
-    if social_engagement_enabled:
-        if course_id and user_id:
-            task_update_user_engagement_score.delay(unicode(course_id), user_id)
+    thread = kwargs['post']
+    course_id = getattr(thread, 'course_id', None)
+    user_id = getattr(thread, 'user_id', None)
 
-        # present only in two cases: thread_deleted & comment_deleted
-        if 'involved_users' in kwargs:
-            involved_users = kwargs['involved_users']
-            users = User.objects.filter(username__in=involved_users).exclude(id=user_id)
+    # present if thread_deleted
+    if 'involved_users' in kwargs:
+        users = kwargs['involved_users']
+        for user, user_data in users.items():
+            _decrement(user, course_id, user_data)
 
-            for user in users:
-                task_update_user_engagement_score.delay(unicode(course_id), user.id)
+    # thread or comment voted
```

Comment:
Don't we no longer check for comment signals on this function?

Reply:
@UmanShahzad, apparently `thread_voted` and `comment_voted` signals are both sent while voting on a comment. I quickly looked at this, but I didn't find the cause of the issue, so I didn't dig further into it - I wasn't sure if it's still in the scope of this ticket and I must have forgotten to tell you about it when I was writing about the other bug.

Reply:
Ah, well that's alright. Yes, it can be left for another ticket.